modules/highgui/src/window_cocoa.mm

```diff
@@ -944,8 +944,8 @@ - (void)createSliderWithName:(const char *)name maxValue:(int)max value:(int *)v
     [[slider name] setStringValue:cvname];
     [[slider slider] setMaxValue:max];
     [[slider slider] setMinValue:0];
-    [[slider slider] setNumberOfTickMarks:(max+1)];
-    [[slider slider] setAllowsTickMarkValuesOnly:YES];
+    [[slider slider] setNumberOfTickMarks:(max > 100 ? 101 : max+1)];
+    [[slider slider] setAllowsTickMarkValuesOnly:NO];
     if(value)
```

Comment:
Can we just hide all ticks? (use 0 instead of 101)

```
[[slider slider] setNumberOfTickMarks:(max > 100 ? 0 : max+1)];
```

Reply:
Yes! I think this is a better solution. You'd mentioned[ limiting the number of ticks as a preferred solution](https://github.com/opencv/opencv/issues/21014#issuecomment-962416058) previously.

No ticks is simpler and solves the problem in a way that's less likely be reverted.

Reply:
Actually I mean hiding for case with more than 100 values:

```.diff
-[[slider slider] setNumberOfTickMarks:(max > 100 ? 101 : max+1)];
+[[slider slider] setNumberOfTickMarks:(max > 100 ? 0 : max+1)];
```

Or

```
if (max <= 100)
{
    [[slider slider] setNumberOfTickMarks:(max+1)];
}
```

---

Please take a look on `cvSetTrackbarMax()` too. We have `setNumberOfTickMarks` call there too.

```
[[slider slider] setNumberOfTickMarks:((max - min) > 100 ? 0 : (max - min +1))];
```

Note: we need to call `setNumberOfTickMarks:0` here if we want to hide ticks.

---

It makes sense to define constant in this file somewhere to avoid using of "magic" number 100 in multiple places (OPENCV_NSSLIDER_MAX_TICKS=100)